### ISSUE TITLE
update doxygen version

### ISSUE THREAD

**Issue by burlen** [readthedocs/readthedocs.org#8151]
## Details
* Read the Docs project URL: https://readthedocs.org/projects/teca
* Build URL (if applicable): https://readthedocs.org/projects/teca/builds/13651770/
* Read the Docs username (if applicable):

## Expected Result
With the versions of Doxygen that ship with Ubuntu 20.04 and other recent OS's Doxygen parses our code correctly.

## Actual Result
On readthedocs.org (and other Ubuntu 18.04 hosts) Doxygen crashes on our code with the following error:
```
Preinput buffer overflow, can't enlarge buffer because scanner uses REJECT
```
See: https://readthedocs.org/api/v2/build/13651770.txt

We desperately need to have the version of doxygen updated in our readthedocs.org environment or to be running on a newer OS image such as 20.04. Can you please help us accomplish one of these?

**Comment by burlen** [comment]
I see that Ubuntu 20.04 is available here: https://github.com/readthedocs/readthedocs-docker-images

I think that will solve my problem but I don't know how to configure our readthedocs.org project to use that version of Ubuntu. Is it possible?

**Comment by astrojuanlu** [comment]
Hi @burlen, thanks for using Read the Docs!

You should be able to use that image by putting

```
build:
  image: testing
```

in your `.readthedocs.yaml`, see https://docs.readthedocs.io/en/stable/config-file/v2.html#build-image

**Comment by burlen** [comment]
@astrojuanlu Thanks for the quick reply. That's great news!

I tried adding the build section as suggested, Unfortunately no change. I'm pasting my yml below. Also, I've added lines to my `conf.py` to `cat /etc/issue` and `doxygen --version` to verify which OS and Doxygen we are getting. After adding the build section, we are still on `Ubuntu 18.04.5 LTS` and Doxygen `1.8.13`.

Let me know if you have any idea why id didn't work?

```
version: 2

build:
  image: testing

sphinx:
  configuration: doc/rtd/conf.py

python:
  version: 3.8
  install:
    - requirements: doc/rtd/requirements.txt

```

**Comment by stsewd** [comment]
Looks like our settings are one version behind https://github.com/readthedocs/readthedocs.org/blob/c941723a510c0d365fafdccb4f83b23562af57c4/readthedocs/settings/base.py#L494-L496

/cc @humitos

**Comment by humitos** [comment]
@stsewd Hrm, I also thought we were using `8.0` (running Ubuntu 20.04) for `testing` at this point. I think we prepared everything to do the upgrade (that's why [README](https://github.com/readthedocs/readthedocs-docker-images/blob/master/README.rst) says that) but then we realized that it could break builds, we started talking about the [new docker image structure](https://github.com/readthedocs/readthedocs.org/pull/7566) to avoid these breaks but we haven't implemented that structure yet.

**Comment by drewcassidy** [comment]
Is there any ETA on a 20.04 docker image being available?

**Comment by burlen** [comment]
Yes, an ETA would be very helpful for planning purposes. It would be extremely helpful if this issue (or is there another one I should watch?) were updated when 20.04 is available.

**Comment by burlen** [comment]
This issue still needs to be resolved, because of the old doxygen/Ubuntu version we are running it manually and uploading a tarball. Very much less than ideal. I had hoped for a quick resolution to this since it had seemed like a newer ubuntu version was in the works. any progress? eta?

**Comment by stsewd** [comment]
We are planning on releasing a new docker image in the following weeks, but not sure if it includes a new testing image as well, /cc @humitos

**Comment by burlen** [comment]
@stsewd thanks for the update. The newer doxygen package versions in 20.04 will resolve my issue. I don't expect any breakages re other packages/versions. I am happy to switch as soon as it is available.

**Comment by humitos** [comment]
> We are planning on releasing a new docker image in the following weeks, but not sure if it includes a new testing image as well, /cc @humitos

We are still discussing how to do the rollout with a newer Ubuntu version. I don't think we will upgrade `testing` to Ubuntu 20.04 LTS, tho. We should implement https://docs.readthedocs.io/en/stable/development/design/build-images.html directly, instead.

**Comment by vhirtham** [comment]
@burlen No need to wait for a new image with Ubuntu `20.04`. I also had issues with my documentation that builds fine on my local machine with version `1.8.17` but isn't compatible with Doxygen `1.8.13`. However, I found a simple solution:
Since Doxygen is now available on conda-forge and readthedocs has conda support, you can use all newer doxygen versions (`1.8.13 ` up to `1.9.1`) in your RTD build. Just add the following line to your RTD config file:

~~~ yaml
conda:
  environment: <DIRECTORY>/environment.yml
~~~

Now create the file `<DIRECTORY>/environment.yml` in your repo with the following content:

~~~ yaml
name: RTD
channels:
  - conda-forge
  - defaults
dependencies:
  - python=3.8
  - doxygen=<VERSION>
~~~

RTD should now build your Doxygen documentation using the version you specified. :wink:

**Comment by njzjz** [comment]
> Since Doxygen is now available on conda-forge and readthedocs has conda support...

Got a memory error, though...

```
Command killed due to excessive memory consumption
```

**Comment by astrojuanlu** [comment]
@njzjz Please send us a support request to enable mamba for your project :)

**Comment by burlen** [comment]
@astrojuanlu @humitos This worked for me! Thank you so much!
